crates/router/src/routes/app.rs

```diff
@@ -342,12 +342,13 @@ impl AppState {
 
             #[allow(clippy::expect_used)]
             #[cfg(feature = "olap")]
-            let opensearch_client = Arc::new(
-                conf.opensearch
-                    .get_opensearch_client()
-                    .await
-                    .expect("Failed to create opensearch client"),
-            );
+            let opensearch_client = conf
+                .opensearch
+                .get_opensearch_client()
+                .await
+                .ok()
```

Comment:
By calling `ok()`, we're losing out on the error information. We can still consider using `expect()` here, or at least log the error using `inspect_err()`. 

If we proceed with not using `expect()`, please remove the `#[allow(clippy::expect_used)]` annotation.